### DISCUSSION TITLE
Is it possible to use toolkit in Node.js ESM with named imports?

### DISCUSSION THREAD

**Discussion by krutoo** [reduxjs/redux-toolkit#3756] [Q&A] [answered]
Hi, i try to use redux toolkit in Node.js v16.15.1 in mjs file by named imports like this:

```js
// index.mjs
import { createAction } from '@reduxjs/toolkit';

createAction('test');
```

and it causes error:

```
Named export 'createAction' not found. The requested module '@reduxjs/toolkit' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@reduxjs/toolkit';
const { createAction, createSelector } = pkg;
```

Is it possible to use redux toolkit in Node.js with named import?

**Comment by markerikson** [comment, marked as answer]
Yes, we've fixed that in RTK 2.0 beta.  (RTK 1.9 will not have those fixes - they need to go in a major version.)

https://github.com/reduxjs/redux-toolkit/releases/tag/v2.0.0-beta.2
